components/user-mgt/org.wso2.carbon.identity.user.profile/src/main/java/org/wso2/carbon/identity/user/profile/mgt/association/federation/FederatedAssociationManagerImpl.java

```diff
@@ -94,6 +94,18 @@ public void createFederatedAssociationWithIdpResourceId(User user, String idpId,
         createFederatedAssociation(user, idpName, federatedUserId);
     }
 
+    @Override
+    public String getAssociatedLocalUser(String tenantDomain, String idpId, String federatedUserId)
```

Comment:
Any reason to pass String instead of User object?

Reply:
Fixed with https://github.com/wso2/carbon-identity-framework/pull/3586/commits/09ce98036625b1efbd3e4220fca37d548c4d2141